app/api/answers/get-answers.go

```diff
@@ -0,0 +1,168 @@
+package answers
+
+import (
+	"errors"
+	"fmt"
+	"github.com/France-ioi/AlgoreaBackend/app/auth"
+	"github.com/go-chi/render"
+	"net/http"
+	"strconv"
+
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+func (srv *Service) getAnswers(rw http.ResponseWriter, httpReq *http.Request) service.APIError {
+	user := srv.GetUser(httpReq)
+
+	dataQuery := srv.Store.UserAnswers().WithUsers().
+		Select(`users_answers.ID, users_answers.sName, users_answers.sType, users_answers.sLangProg,
+            users_answers.sSubmissionDate, users_answers.iScore, users_answers.bValidated,
+            users.sLogin, users.sFirstName, users.sLastName`).
+		Order("sSubmissionDate DESC")
+
+	userID, userIDError := resolveURLQueryGetInt64Field(httpReq, "user_id")
```

Comment:
Shouldn't this "input validation" be moved in a separate function?  The part checking the args, up to `either user_id & item_id or attempt_id must be present. Maybe that will not be much better, I let you think about it.

Reply:
I think in this case a separate function won't help much. It can make things even more complex